### ISSUE TITLE
With "Sync on start": Decryption error on program start (correct password entered) when "Sync on start" is activated

### ISSUE THREAD

**Issue by Wikinaut** [ola-ct/Qt-SESAM#56]
With "Sync on start" I got a Decryption error window on program start (correct password entered).

**Comment by 607011** [comment]
Does this issue anything have to do with #57 ?

**Comment by Wikinaut** [comment]
No, I filed this separately, because I had problems now and then.

What did I do?

"on the fly" I changed the name of the sync file (in Extras > Options...), and copied the file to the new location.

Made some changes (because otherwise the SAVE button was grey and settings could not be stored - this may be subject of another issue).
Closed Qt-Sesam then.

Tried to restart Qt-Sesam.
Error message comes.

In short:
there is a lurking problem with the sync file filename. When the sync filename was changed, a new start is not possible.

I am sorry that I don't have further details at the moment.

**Comment by Wikinaut** [comment]
Happens with "Sync on Start".

**Comment by Wikinaut** [comment]
Sorry, correcting this:

**Comment by Wikinaut** [comment]
Happens WITHOUT changing the sync file. Only activated the "Sync on Start" - restart - bug

**Comment by 607011** [comment]
Thanks. I'll dig into it.

**Comment by 607011** [comment]
@Wikinaut Presumed there is a valid (i.e. decryptable) domain settings file set via Extras/Options, you have to do the following to move this file to another location:
- Start QtSESAM.
- Open the options dialog (Extras/Options).
- In Synchronize tab select a new location for the sync file.
- Move the file to the selected location.
- Close QtSESAM
- Start QtSESAM.

This works whether or not Sync On Start is enabled.

**Comment by 607011** [comment]
@Wikinaut wrote: "When the sync filename was changed, a new start is not possible." 

Do you expect that QtSESAM is capable of guessing what happened to a file it needs for proper operation? ;-)

Of course, QtSESAM should display a more descriptive error message in such cases. I'm working on it. Thanks for your help and patience.

**Comment by Wikinaut** [comment]
@ola-ct  of course, I also moved the file. Let#s concentrate to the "Sync on start" problem (with fixed filename). Can you try to reproduce it ?
- Setup a fresh installation
- setup a domain entry
- close QTS
- restart QTS
- set "Sync on start"
- close QTS
- start QTS - now it shows the error message

**Comment by 607011** [comment]
It doesn't show an error message because your steps don't include choosing the name of a file to sync with ;-)

So I did the following with a fresh installation:
- Start QtSESAM.
- Set up a domain entry
- Choose sync file.
- Select Sync With File.
- Select Sync On Start.
- Close QtSESAM.
- Start QtSESAM.

The error appears.

I confirm that this is a bug :-(

**Comment by 607011** [comment]
If you synchronise before you close QtSESAM for the first time, the error does not appear. So I guess this happens because the sync peer has no data.

**Comment by Wikinaut** [comment]
Negative.

As long as I have "Sync on start" activated, I cannot access the sync file (the password error message comes up).

The sync file is located in a folder, which is connected (synchronised every minute) with a file on my owncloud server. It does not matter, whether or not the sync client is really started. This is why I suppose there's an error in the code related to the "sync on start" setting.

**Comment by 607011** [comment]
@Wikinaut Seems to be fixed with commit 129d06e39b11f17fc30460eaeed46010d9a23388. Can you confirm that?

**Comment by Wikinaut** [comment]
WTF - cannot access any more now

Error message comes always,cannot go to the program to reset the flag

**Comment by Wikinaut** [comment]
#SyncFlagGate ;-)

**Comment by 607011** [comment]
No problems here, neither with a clean installation nor with an existing, heavily used one.

**Comment by Wikinaut** [comment]
The sync file is in a folder, which is sync'd with ownCloud ? On Linux ?

**Comment by 607011** [comment]
My sync file is in a MS OneDrive folder. Using Windows 7.

**Comment by 607011** [comment]
Also tested with Linux, sync file in a local folder not managed by a cloud service. Works fine.

**Comment by Wikinaut** [comment]
The behaviour is different between 4b0211880d0c10ad9a63c9ebec727a6dc3e732d9 and 8d0f67099d524bbd0dd4af822c565ebdf166be6a .

The new version is unusable. The older version shows the error window, but closing this let me got to the main window and let me reset the Sync-on-start flag. The newer version show the error window, closing it goes back to the password window, then again the error window etc.

Unusable. Sorry.

**Comment by Wikinaut** [comment]
Unusable in case "Sync on start"

**Comment by Wikinaut** [comment]
I have a i5 4250U . Perhaps too quick?

**Comment by 607011** [comment]
We've played around a lot with "sync on start" enabled and disabled, with pristine and used installations, with Windows and Linux, on fast machines and in VMware, and could not find a bug that led to the behavior you described, @Wikinaut. So I declare this issue resolved by commit 4ca7f907614d13c1c1c01e9ba4f07e8252614949.

**Comment by Wikinaut** [comment]
<img src="http://i.imgur.com/IBZJFxa.png">

Problem still present in 9e1ae374e7e5a62f84312368f6b3887da1e26762 . Giving up with your program.

**Comment by 607011** [comment]
I'm sure there's a corrupt settings or sync file remaining.

With settings file I mean QtSESAM.ini residing in ~/.config/ct. With sync file I mean the one you set in "Extras/Options/Sync file".

Please remove them.

Then try again. TIA!

**Comment by Wikinaut** [comment]
Before compiling from scratch I removed the sync file. I will now also remove the ini file. 

By the way
1.  _where_ is the sync-on-start flag stored ?
2.  A description whay "sync on start" actually does, and why it is useful, or not, should be in a -> tooltip

**Comment by Wikinaut** [comment]
The program is not generating a password. Spinner is shown, no characters in the password field.

**Comment by Wikinaut** [comment]
There is something wrong with the code.

**Comment by 607011** [comment]
"Spinner is shown, no characters in the password field." - Yes, I've also seen this in rare cases. Retry selecting a domain from the combobox for intermediate remedy.

**Comment by 607011** [comment]
The "sync-on-start flag" is stored encrypted with the other sync settings in QtSESAM.ini and therein `sync/params`.

**Comment by 607011** [comment]
Tooltip for "sync on start": noted!

**Comment by Wikinaut** [comment]
> "Spinner is shown, no characters in the password field." - Yes, I've also seen this in rare cases. Retry selecting a domain from the combobox for intermediate remedy.

It comes, reproducible, after fresh program setup. "New domain" > domain entered > "Accept". Again, there is big bug lurking somewhere. try to build a fresh code from scratch,

**Comment by 607011** [comment]
@Wikinaut, could you please do

```
git pull origin Issue-68
git checkout Issue-68
```

then build and test if the bug (now documented as issue #68) is still present? Thanks in advance.

**Comment by Wikinaut** [comment]
same problem. endless-loop as reported. Also, no password generated as reported. Giving up.

**Comment by Wikinaut** [comment]
Still present https://github.com/ola-ct/Qt-SESAM/issues/56

**Comment by Wikinaut** [comment]
I decided now to try a different file location.

The file which made trouble was on /home/someuser/.local/share/QtSesam.bin - this appears not be possible, I don't know why. I chose now for /home/someuser/owncloud/something/somefilename.bin .

So far no problems.

**Comment by Wikinaut** [comment]
Arrrghhh, and this problem is **back** "Decryption error". Cannot use your program.
